tests/Integration/MediaWikiNormalizedTermCleanerTest.php

```diff
@@ -0,0 +1,281 @@
+<?php // phpcs:disable PSR1.Files.SideEffects.FoundWithSymbols -- see evil hack below
+
+namespace Wikibase\TermStore\MediaWiki\Tests\Integration;
+
+use MediaWikiTestCase;
+use Wikibase\TermStore\MediaWiki\PackagePrivate\MediaWikiNormalizedTermCleaner;
+use Wikimedia\Rdbms\IMaintainableDatabase;
+
+// Evil hack because we don’t always have MediaWikiTestCase and `class extends MediaWikiTestCase`
+// crashes even if (because of the special group) we’re not actually going to run the test.
+// This also requires us to disable the PSR1.Files.SideEffects.FoundWithSymbols warning, which
+// otherwise complains about mixing declarations with procedural code; since that warning is
+// reported on line 1, not on the return statement, we have to disable it up there instead of here.
+if ( !class_exists( MediaWikiTestCase::class ) ) {
+	return;
+}
+
+/**
+ * @covers \Wikibase\TermStore\MediaWiki\PackagePrivate\MediaWikiNormalizedTermCleaner
+ *
+ * @group Database
+ * @group TermStoreWithMediaWikiCore
+ *
+ * @license GPL-2.0-or-later
+ */
+class MediaWikiNormalizedTermCleanerTest extends MediaWikiTestCase {
+
+	protected function setUp() {
+		parent::setUp();
```

Comment:
I avoided using MediaWikiTestCase in here https://github.com/wmde/mediawiki-term-store/pull/11/files#diff-e081a73b437637a0cbe8c8b34c40e388R23, as I do not need any of the logic about tables copying and the rest of that logic .. there are no dependencies on anything mediawiki on db level here, so I suggest changing this to how it is in the other PR to keep it consistent ?

We can also eventually introduce a base test case class in this lib that does what the other PR does.. I'd prefer that actually, but can be done later once both are in master as a maintenance follow-up

Reply:
I tried to copy your approach, but couldn’t figure out how I need to manipulate the sqlite database so that the `beginMasterChanges()` doesn’t cause an error, so I’ll stick to `MediaWikiTestCase` for now and we can try improving it later.